ui/pages/send/send-content/send-content.component.js

```diff
@@ -77,11 +121,90 @@ export default class SendContent extends Component {
       recipient.warning === 'knownAddressRecipient';
     const hideAddContactDialog = recipient.warning === 'loading';
 
+    let title;
+    if (
+      draftTransaction?.asset.details?.standard === ERC721 ||
+      draftTransaction?.asset.details?.standard === ERC1155
+    ) {
+      title = draftTransaction?.asset.details?.name;
+    } else if (draftTransaction?.asset.details?.standard === ERC20) {
+      title = `${hexWEIToDecETH(draftTransaction.amount.value)} ${
+        draftTransaction?.asset.details?.symbol
+      }`;
+    }
+
+    const ethTransactionTotalMaxAmount = Number(
+      hexWEIToDecETH(hexMaximumTransactionFee),
+    );
+
+    const primaryTotalTextOverrideMaxAmount = `${title} + ${ethTransactionTotalMaxAmount} ${nativeCurrency}`;
+
+    const detailText = () => {
+      return (
+        <Box
+          height={BLOCK_SIZES.MAX}
+          display={DISPLAY.FLEX}
+          flexDirection={FLEX_DIRECTION.COLUMN}
+          className="gas-display__total-value"
+        >
+          <LoadingHeartBeat estimateUsed={draftTransaction?.userFeeLevel} />
+          <UserPreferencedCurrencyDisplay
+            type={SECONDARY}
+            key="total-detail-text"
+            value={hexTransactionTotal}
+            hideLabel={Boolean(useNativeCurrencyAsPrimaryCurrency)}
+          />
+        </Box>
+      );
+    };
+
+    const detailTotal = () => {
+      if (draftTransaction?.asset.type === 'NATIVE') {
+        return (
+          <Box
+            height={BLOCK_SIZES.MAX}
+            display={DISPLAY.FLEX}
+            flexDirection={FLEX_DIRECTION.COLUMN}
+            className="gas-display__total-value"
+          >
+            <LoadingHeartBeat estimateUsed={draftTransaction?.userFeeLevel} />
+            <UserPreferencedCurrencyDisplay
+              type={PRIMARY}
+              key="total-detail-value"
+              value={hexTransactionTotal}
+              hideLabel={!useNativeCurrencyAsPrimaryCurrency}
+            />
+          </Box>
+        );
+      }
+      return useNativeCurrencyAsPrimaryCurrency
+        ? primaryTotalTextOverrideMaxAmount
+        : undefined;
+    };
+
+    const maxAmount = () => {
+      if (draftTransaction?.asset.type === 'NATIVE') {
+        return (
+          <UserPreferencedCurrencyDisplay
+            type={PRIMARY}
+            key="total-max-amount"
+            value={addHexes(
+              draftTransaction.amount.value,
+              hexMaximumTransactionFee,
+            )}
+            hideLabel={!useNativeCurrencyAsPrimaryCurrency}
+          />
+        );
+      }
+      return useNativeCurrencyAsPrimaryCurrency
+        ? primaryTotalTextOverrideMaxAmount
+        : undefined;
+    };
```

Comment:
these should not be functions. They should either be inline in the props, or just variables.

Reply:
@brad-decker Done [here](https://github.com/MetaMask/metamask-extension/pull/14634/commits/3a34638c9acaf99d1d84076857a756c6b30c0a01).